src/mscorlib/shared/System/String.Searching.cs

```diff
@@ -63,24 +66,35 @@ public int IndexOf(char value, StringComparison comparisonType)
 
                 case StringComparison.OrdinalIgnoreCase:
                     return CompareInfo.Invariant.IndexOf(this, value, CompareOptions.OrdinalIgnoreCase);
-                    
+
                 default:
                     throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
             }
         }
-        
+
         public unsafe int IndexOf(char value, int startIndex, int count)
         {
-            if (startIndex < 0 || startIndex > Length)
+            if ((uint)startIndex > (uint)Length)
```

Comment:
>f ((uint)startIndex > (uint)Length) [](start = 13, length = 35)

wouldn't this can generate a problem or a different exception if passing startIndex as negative value?

Reply:
Nope.

> If start was negative, casting it to uint would make it larger than Int32.MaxValue and would automatically be larger than segment.Count. So the check is already there (just optimized).


We have such checks everywhere.
See https://github.com/dotnet/coreclr/pull/16658#discussion_r171327749

Reply:
if someone calling this API inside unchecked block would result in undesired results I guess: 

for example 
            Console.WriteLine(unchecked((int)-4294967294));
is printing 
2



---
In reply to: [171733137](https://github.com/dotnet/coreclr/pull/16392#discussion_r171733137) [](ancestors = 171733137)

Reply:
this is not true if using unchecked blocks. 
for example 
            Console.WriteLine(unchecked((int)-4294967294));
is printing 
2

---
In reply to: [171733480](https://github.com/dotnet/coreclr/pull/16392#discussion_r171733480) [](ancestors = 171733480)

Reply:
never mind, the value would be just 2. I think this will work I guess

---
In reply to: [171734390](https://github.com/dotnet/coreclr/pull/16392#discussion_r171734390) [](ancestors = 171734390,171733480)

Reply:
never mind, the value would be just 2. I think this will work I guess. 

could you add some comment mentioning the limits and how this is safe?

---
In reply to: [171733980](https://github.com/dotnet/coreclr/pull/16392#discussion_r171733980) [](ancestors = 171733980,171733137)

Reply:
at least adding the comment on the newly introduced code.

Reply:
There are hundreds of places in coreclr/corefx that use this idiom. IMHO, I do not think that the comment is needed.